portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -71,108 +72,86 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
-            3. user specifies amount for all messages under a list of tags
-        */ 
-
     }
     
-
-    /**
-   * Converts a ServerStats instance into a JSON string using the Gson library
-   */
-    @Override
+   @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      // Get Post parameters.
       String message = request.getParameter("text-input");
-
-      String nickname = getParameter(request, "sender", "Steven");
-
-      numberOfCommentsToDisplay = getNumberOfCommentsToDisplay(request);
+      String nickname = request.getParameter("sender");
+      String loadFactor = request.getParameter("load_factor");
+      String postTag = getTagParameter(request, "tags", InternalTags.defaultTag());
       
-      // TODO: 
-      //      Get default tag from the InternalTags class and use that below.
+      //TODO: Handle image file sent with FormData.
+      //String imageUrl = getUploadedFileUrl(request, imagePath);
       
-      // Get type of comment.
-      String commentType = getParameter(request, "tags", "Default");
-
-      String imageUrl = getUploadedFileUrl(request, "image");
+      // Only put BlogMessages with a message and tag in datastore.
+      if (message.equals("") || message.equals(null) || postTag == null) {
+        return;
+      }
 
-      String messageRepliesString = getParameter(request, "replies", "");
-	    String messageRepliesArray[] = messageRepliesString.split(",");
-	    List<String> messageReplies = new ArrayList<String>();
-	    messageReplies = Arrays.asList(messageRepliesArray);
-      
       long timestamp = System.currentTimeMillis();
-
       Entity blogMessageEntity = new Entity("blogMessage");
       blogMessageEntity.setProperty("nickname", nickname);
       blogMessageEntity.setProperty("text", message);
-      blogMessageEntity.setProperty("imgUrl", imageUrl);
+      // blogMessageEntity.setProperty("imgUrl", imageUrl);
       blogMessageEntity.setProperty("time", timestamp);
-      blogMessageEntity.setProperty("tag", commentType);
+      blogMessageEntity.setProperty("tag", postTag);
       blogMessageEntity.setProperty("replies", messageReplies);
+    
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       datastore.put(blogMessageEntity);
 
-      response.sendRedirect("/index.html");
+      // To get the recently posted message, add its tag to the |tagsToSearch| list.
+      // If the list has tags already, clear them before adding the tag.
+      // Our goal is to get the recently posted message.
+      if (!tagsToSearch.isEmpty()) {
+        tagsToSearch.clear();
+      }
+      tagsToSearch.add(postTag);
+
+      // Everytime a post comes through, the javascript function that handles the Post assigns
+      // a load_factor value of 1.
+      // We assign that to the |numberOfCommentsToDisplay|. 
+      if (loadFactor != null) {
```

Comment:
I don't really understand this section. You say that loadFactor will always be 1, so why shouldn't we just set numberOfCommentsToDisplay to 1?

Reply:
oh i meant to say "will always be a number". The previous implementation had 2 post functionalities, each of which required a different numberOfCommentsToDisplay value. so the idea was to assign the value that each post function required to the numberOfCommentsToDisplay.
But when i took away one of the post functions i forgot to make the change here to simply set numberOfCommentsToDisplay to 1.
So you're right I can just set it to 1 now.